Doc/whatsnew/3.13.rst

```diff
@@ -206,6 +206,14 @@ array
 ast
 ---
 
+* The constructors of node types in the :mod:`ast` module are stricter
+  in the arguments they accept. If an optional field is not passed to the
+  constructor, it is set to ``None``, and if a list field is not passed,
+  it is set to the empty list. (Previously, the attribute would be missing.)
+  If any other field is not passed to the constructor, a :exc:`DeprecationWarning`
+  is raised. This will raise an exception in Python 3.15. Similarly, passing
+  a keyword argument that does not map to a field on the AST node now raises
+  a :exc:`DeprecationWarning` and will raise an exception in Python 3.15.
```

Comment:
This all feels a bit terse and hard-to-understand to me right now. I like the wording you put in `ast.rst` more.

Maybe something like this? Could possibly consider a short REPL example as well

```suggestion
* The constructors of node types in the :mod:`ast` module are now stricter
  in the arguments they accept, and have more intuitive behaviour when
  arguments are omitted.

  If an optional field on an AST node is not included as an argument when
  calling that node's constructor method, the field will now be set to
  ``None``. Similarly, if an AST node has a list field and that field is
  omitted when calling the node's constructor method, that field will now be
  set to an empty list. (Previously, in both cases, the attribute would be
  missing on the newly constructed AST node instance.)

  If there are fewer arguments passed to an AST node constructor than there are
  fields on that AST node, a :exc:`DeprecationWarning` will now be emitted.
  Passing too few arguments to the constructor will cause an exception in
  Python 3.15. Similarly, passing a keyword argument that does not map to a
  field on the AST node is now deprecated, and will cause an exception to be
  raised in Python 3.15.
```

Reply:
I suspect we should not use the suggested wording here in the third para "fewer arguments passed than there are fields on the node," as that wording is not really true (it fails to consider optional and list nodes.) We could maybe say "fewer arguments passed than there are _required_ fields on the node," but I still think that's less clear than just saying "if no value is provided for a required argument..."

Otherwise (for the first two paras) I find the suggested wording fine, probably a bit clearer, also maybe a bit more verbose than necessary in a few spots (do we really need to say "Similarly, if an AST node has a list field and that field is omitted..." rather than just "Similarly, f a list field is omitted..."?) But I generally find either the original or suggested text basically fine.

Reply:
Thanks both. I pushed a version that is based on Alex's wording, but with the correction Carl brought up and edited for conciseness.